### ISSUE TITLE
`insert` should append a value?

### ISSUE THREAD

**Issue by amitaibu** [pzp1997/assoc-list#4]
Currently we prepend the new value and it appears on the top of the list.

https://github.com/pzp1997/assoc-list/blob/63ffd69a46b7805dba9c3a9ab82c5d07a8bd57a6/src/AssocList.elm#L191

This means that if we have a list

`[ ( 1, 1 ), ( 2, 2 ), ( 3, 3 ), ( 4, 4 ) ] |> Dict.fromList`

results with  `D [(4,4),(3,3),(2,2),(1,1)]`

I'd expect the list order to remain as is, or is there a good reason not to?

**Comment by amitaibu** [comment]
Another good example is 

`([ ( 1, 1 ), ( 2, 2 ), ( 3, 3 ), ( 4, 4 ) ] |> Dict.fromList |> Dict.toList`

ends up with `[(4,4),(3,3),(2,2),(1,1)]`

**Comment by pzp1997** [comment]
I agree with you that the behavior is a bit surprising. This was one of the aspects of the library that I was the most unsure about before publishing. The reason `fromList` reverses the list is because it uses `List.foldl` rather than `List.foldr`. At the time this seemed like the right choice to me for two reasons, `List.foldl` is a bit faster and it is what elm/core `Dict` uses in its `fromList` implementation. In hindsight, this decision feels more and more like a mistake to me.

With regards to changing `insert`, that is not a great way to solve this problem since appends are _way_ slower than conses both in theory and in practice. We could do a trick by writing a combined filter-and-append operation using `List.foldr` but it seems like more trouble than its worth given that there's an even simpler fix involving `fromList`. In short, the "problem" is with `fromList` not `insert`.

I want to make the change to `fromList` but might need to wait for another breaking change so I can batch them and trick Elm's automatic semver enforcement into accepting this as a major bump.

**Comment by amitaibu** [comment]
Hi, just checking if there's a plan to change this soon? (currently I'm running a local patched lib, but would be happy to use upstream)

**Comment by pzp1997** [comment]
I am going to try to release a new version once I am done with school in about two weeks and I expect this change to be included. Hang in there!

**Comment by amitaibu** [comment]
Hi :wave: 

Anything I can help to get it forward? I just added the forked file to the 4th Gizra project, and I feel bad about it every time!  :wink:

**Comment by amitaibu** [comment]
Hi @pzp1997 are you still interested in this fix? We're now at the ~7th project with our own fork.
